src/main/java/Test.java

```diff
@@ -0,0 +1,32 @@
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class Test {
+    public void calculateFunction(Long principal) {
```

Comment:
Method parameter 'principal' might be null - consider adding null check

Reply:
resolved